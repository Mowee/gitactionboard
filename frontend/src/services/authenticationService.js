const getCookies = () =>
  document.cookie.split(';').map((cookie) => cookie.trim());

const getCookie = (name) => {
  const processedName = `${name}=`;
  const cookies = getCookies();

  for (const cookie of cookies) {
    if (cookie.indexOf(processedName) === 0) {
      return decodeURIComponent(cookie).substring(
        processedName.length,
        cookie.length
      );
    }
  }
};

export const clearCookies = () => {
  const cookies = getCookies();
  for (const item of cookies) {
    const indexOfEqual = item.indexOf('=');
    const name = indexOfEqual > -1 ? item.substring(0, indexOfEqual) : item;
    document.cookie = name + '=;expires=Thu, 01 Jan 1970 00:00:00 GMT';
  }
};

export const fetchAccessToken = () => getCookie('access_token');

export const isAuthenticate = () => !!fetchAccessToken();

export const getName = () => getCookie('name') || 'Guest';

export const getAvatarUrl = () => getCookie('avatar_url') || '';
