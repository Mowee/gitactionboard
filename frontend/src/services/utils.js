import storageService from '@/services/storageService';
import preferences from '@/services/preferences';

const ALL_POSSIBLE_EVENTS = {
  branch_protection_rule: 'Branch protection rule',
  check_run: 'Check run',
  check_suite: 'Check suite',
  create: 'Create',
  delete: 'Delete',
  deployment: 'Deployment',
  deployment_status: 'Deployment status',
  discussion: 'Discussion',
  discussion_comment: 'Discussion comment',
  fork: 'Fork',
  gollum: 'Gollum',
  issue_comment: 'Issue comment',
  issues: 'Issues',
  label: 'Label',
  member: 'Member',
  merge_group: 'Merge Group',
  milestone: 'Milestone',
  page_build: 'Page build',
  project: 'Project',
  project_card: 'Project card',
  project_column: 'Project column',
  public: 'Public',
  pull_request: 'Pull request',
  pull_request_review: 'Pull request review',
  pull_request_review_comment: 'Pull request review comment',
  pull_request_target: 'Pull request target',
  push: 'Push',
  registry_package: 'Registry package',
  release: 'Release',
  status: 'Status',
  watch: 'Watch',
  workflow_call: 'Workflow call',
  workflow_dispatch: 'Workflow dispatch',
  workflow_run: 'Workflow run',
  repository_dispatch: 'Repository dispatch',
  schedule: 'Schedule'
};

const SECOND = 1000;
const MINUTE = 60 * SECOND;
const HOUR = 60 * MINUTE;
const DAY = 24 * HOUR;
const WEEK = 7 * DAY;
const MONTH = 30 * DAY;
const YEAR = 12 * MONTH;

const RELATIVE_FORMAT_INFO = [
  { cutoffs: MINUTE, unit: 'seconds', amount: SECOND },
  { cutoffs: HOUR, unit: 'minutes', amount: MINUTE },
  { cutoffs: DAY, unit: 'hours', amount: HOUR },
  { cutoffs: WEEK, unit: 'days', amount: DAY },
  { cutoffs: MONTH, unit: 'weeks', amount: WEEK },
  { cutoffs: YEAR, unit: 'months', amount: MONTH },
  { cutoffs: Number.POSITIVE_INFINITY, unit: 'years', amount: YEAR }
];

const RELATIVE_TIME_FORMAT = new Intl.RelativeTimeFormat('en', {
  numeric: 'auto',
  style: 'long'
});

export const setAvailableAuths = (availableAuths) => {
  storageService.setItem('availableAuths', JSON.stringify(availableAuths));
};

export const setVersion = (version) => {
  storageService.setItem('version', version);
};

export const setGithubSecretsScanMonitoringEnabled = (
  githubSecretsScanMonitoringEnabled
) => {
  storageService.setItem(
    'githubSecretsScanMonitoringEnabled',
    JSON.stringify(githubSecretsScanMonitoringEnabled)
  );
};

export const setGithubCodeScanMonitoringEnabled = (
  githubSecretsScanMonitoringEnabled
) => {
  storageService.setItem(
    'githubCodeScanMonitoringEnabled',
    JSON.stringify(githubSecretsScanMonitoringEnabled)
  );
};

export const getAvailableAuths = () =>
  JSON.parse(storageService.getItem('availableAuths'));

export const getGithubSecretsScanMonitoringEnabled = () =>
  JSON.parse(storageService.getItem('githubSecretsScanMonitoringEnabled')) ||
  false;

export const getGithubCodeScanMonitoringEnabled = () =>
  JSON.parse(storageService.getItem('githubCodeScanMonitoringEnabled')) ||
  false;

export const getVersion = () => storageService.getItem('version');

const buildTriggeredEvents = (keys) => {
  return keys.map(key => ({
    title: ALL_POSSIBLE_EVENTS[key],
    value: key
  }));
};

export const getAllPossibleTriggeredEvents = () => buildTriggeredEvents(Object.keys(ALL_POSSIBLE_EVENTS));

export const getShowBuildsDueToTriggeredEvents = () =>
  preferences.showBuildsDueToTriggeredEvents.length === 0
    ? Object.keys(ALL_POSSIBLE_EVENTS)
    : preferences.showBuildsDueToTriggeredEvents;

export const getRelativeTime = timestamp => {
  const timeDifference = new Date(timestamp) - new Date();

  const { amount, unit } =
      RELATIVE_FORMAT_INFO.find(({ cutoffs }) => cutoffs > Math.abs(timeDifference)) ||
      RELATIVE_FORMAT_INFO[0];

  return RELATIVE_TIME_FORMAT.format(Math.round(timeDifference / amount), unit);
};
