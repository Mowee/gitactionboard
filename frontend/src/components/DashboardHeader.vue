<template>
  <v-app-bar
    elevation="5"
  >
    <v-row no-gutters>
      <v-col cols="10">
        <v-app-bar-title
          class="header-title text-center font-weight-bold pl-10"
        >
          <span>Gitaction Board</span> <span class="version">(v{{ version }})</span>
        </v-app-bar-title>
      </v-col>
      <v-col
        cols="2"
        class="d-inline-flex"
      >
        <v-app-bar-title
          id="sub-header"
          class="pr-4 text-right font-weight-bold align-center fill-height d-inline-flex flex-row-reverse mt-3"
        >
          {{ subHeader }}
        </v-app-bar-title>
      </v-col>
    </v-row>
  </v-app-bar>
</template>

<script>
import { getVersion } from '@/services/utils';

export default {
  name: 'DashboardHeader',
  props: {
    subHeader: {
      type: String,
      required: true
    }
  },
  data() {
    return {
      version: getVersion()
    };
  }
};
</script>

<style scoped>
.header-title {
    line-height: normal;
    font-size: 50px;
    font-family: "Snell Roundhand", cursive;
}

.version {
    font-size: 15px;
    font-family: "American Typewriter", serif;
}

#sub-header {
    font-size: 20px;
    font-family: "American Typewriter", serif;
}

</style>
