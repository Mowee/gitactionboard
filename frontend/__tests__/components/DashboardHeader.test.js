import { describe, it, expect, vi, afterEach } from 'vitest';
import DashboardHeader from '@/components/DashboardHeader';
import { getVersion } from '@/services/utils';
import { mountWithWrapper } from '../test-utils';

describe('<DashboardHeader />', () => {
  vi.mock('@/services/utils', () => {
    return {
      getVersion: vi.fn().mockReturnValue('3.1.0')
    };
  });

  afterEach(vi.clearAllMocks);

  it.each([
    ['Workflow Jobs'], ['Exposed Secrets'], ['Code Standard Violations']
  ])('should render dashboard header for %s', (subHeader) => {
    expect(mountWithWrapper(DashboardHeader, {
      props: {
        subHeader
      }
    }).html()).toMatchSnapshot();
    expect(getVersion).toBeCalledTimes(1);
  });
});
