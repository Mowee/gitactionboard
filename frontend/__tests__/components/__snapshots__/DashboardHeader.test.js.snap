// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DashboardHeader /> > should render dashboard header for Code Standard Violations 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr" style="z-index: 1000;">
  <div class="v-application__wrap">
    <header class="v-toolbar v-toolbar--density-default elevation-5 v-theme--light v-locale--is-ltr v-app-bar" style="top: 0px; z-index: 1004; transform: translateY(0%); position: fixed; transition: none;">
      <!---->
      <div class="v-toolbar__content" style="height: 64px;">
        <!---->
        <!---->
        <div class="v-row v-row--no-gutters">
          <div class="v-col v-col-10">
            <div class="v-toolbar-title header-title text-center font-weight-bold pl-10 v-app-bar-title">
              <div class="v-toolbar-title__placeholder">
                <!---->
                <span>Gitaction Board</span>
                <span class="version">(v3.1.0)</span>
              </div>
            </div>
          </div>
          <div class="v-col v-col-2 d-inline-flex">
            <div class="v-toolbar-title pr-4 text-right font-weight-bold align-center fill-height d-inline-flex flex-row-reverse mt-3 v-app-bar-title" id="sub-header">
              <div class="v-toolbar-title__placeholder">
                <!---->
                Code Standard Violations
              </div>
            </div>
          </div>
        </div>
        <!---->
      </div>
      <transition-stub appear="false" css="true" name="expand-transition" persisted="false">
        <!---->
      </transition-stub>
    </header>
  </div>
</div>
`;

exports[`<DashboardHeader /> > should render dashboard header for Exposed Secrets 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr" style="z-index: 1000;">
  <div class="v-application__wrap">
    <header class="v-toolbar v-toolbar--density-default elevation-5 v-theme--light v-locale--is-ltr v-app-bar" style="top: 0px; z-index: 1004; transform: translateY(0%); position: fixed; transition: none;">
      <!---->
      <div class="v-toolbar__content" style="height: 64px;">
        <!---->
        <!---->
        <div class="v-row v-row--no-gutters">
          <div class="v-col v-col-10">
            <div class="v-toolbar-title header-title text-center font-weight-bold pl-10 v-app-bar-title">
              <div class="v-toolbar-title__placeholder">
                <!---->
                <span>Gitaction Board</span>
                <span class="version">(v3.1.0)</span>
              </div>
            </div>
          </div>
          <div class="v-col v-col-2 d-inline-flex">
            <div class="v-toolbar-title pr-4 text-right font-weight-bold align-center fill-height d-inline-flex flex-row-reverse mt-3 v-app-bar-title" id="sub-header">
              <div class="v-toolbar-title__placeholder">
                <!---->
                Exposed Secrets
              </div>
            </div>
          </div>
        </div>
        <!---->
      </div>
      <transition-stub appear="false" css="true" name="expand-transition" persisted="false">
        <!---->
      </transition-stub>
    </header>
  </div>
</div>
`;

exports[`<DashboardHeader /> > should render dashboard header for Workflow Jobs 1`] = `
<div class="v-application v-theme--light v-layout v-layout--full-height v-locale--is-ltr" style="z-index: 1000;">
  <div class="v-application__wrap">
    <header class="v-toolbar v-toolbar--density-default elevation-5 v-theme--light v-locale--is-ltr v-app-bar" style="top: 0px; z-index: 1004; transform: translateY(0%); position: fixed; transition: none;">
      <!---->
      <div class="v-toolbar__content" style="height: 64px;">
        <!---->
        <!---->
        <div class="v-row v-row--no-gutters">
          <div class="v-col v-col-10">
            <div class="v-toolbar-title header-title text-center font-weight-bold pl-10 v-app-bar-title">
              <div class="v-toolbar-title__placeholder">
                <!---->
                <span>Gitaction Board</span>
                <span class="version">(v3.1.0)</span>
              </div>
            </div>
          </div>
          <div class="v-col v-col-2 d-inline-flex">
            <div class="v-toolbar-title pr-4 text-right font-weight-bold align-center fill-height d-inline-flex flex-row-reverse mt-3 v-app-bar-title" id="sub-header">
              <div class="v-toolbar-title__placeholder">
                <!---->
                Workflow Jobs
              </div>
            </div>
          </div>
        </div>
        <!---->
      </div>
      <transition-stub appear="false" css="true" name="expand-transition" persisted="false">
        <!---->
      </transition-stub>
    </header>
  </div>
</div>
`;
